File: Difficulty/Medium/[209] Minimum Size Subarray Sum/solution.py
class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        if s <= 0 or not nums: return 0
        left, right, sum = 0, 0, 0
        length, minLen = len(nums), 2**31 - 1
        while right < length:
            sum += nums[right]
            right += 1
            while sum >= s:
                minLen = min(minLen, right - left)
                sum -= nums[left]
                left += 1
        return 0 if minLen == 2 ** 31 - 1 else minLen 

File: Difficulty/Medium/[209] Minimum Size Subarray Sum/solution.cpp
/*
 * @lc app=leetcode id=209 lang=cpp
 *
 * [209] Minimum Size Subarray Sum
 */

// @lc code=start
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        if (s <= 0 || nums.empty()) return 0;
        int left = 0;
        int right = 0;
        int sum = 0;
        int length = nums.size();
        int minLen = INT_MAX;
        while (right < length) {
            sum += nums[right];
            right++;
            while (sum >= s) {
                minLen = min(minLen, right - left);
                sum -= nums[left];
                left++;
            }
        }
        return minLen == INT_MAX ? 0 : minLen;
    }
};
// @lc code=end

